content/docs/plugin-docs/json/presetsv2.md

```diff
@@ -0,0 +1,228 @@
+
+
+##### Input Properties
+
+Input Properties are poperties that can be defined inside of an input box these are grouped in a `object` and will change depending on what input type is selected. There are 3 properties that work with all input types and those are:
+
+- `type` Type of input.
+- `default` Default value for the input.
+- `optional` Whether or not the input is optional.
+
+
+### Icon
+Icons show up in the presets window. Presets use [Material Design Icons](https://materialdesignicons.com/) a collection of icons that can be defined in the manifest.json by typing `mdi-` followed by the icon's name.
+
+
+### Variables
+
+Variables are able to be used in the `createFiles` and `expandFile` components of the `manifest.json they` can also be used any other files in the preset as long as they are "injected" in the file.
+
+Variables can be referenced by using them inside of two curly brackets: `{{VARIABLE}}`. bridge. automatically replaces variables with the current corresponding variable value.
+
+Note: The variable `PROJ_PREFIX` is already pre-defined and contains the namespace of the project (colon is not included).
+
+
+
+
+## Examples
+
+### Manifest Example
+
+```javascript
+{
+	"name": "Bat",
+	"icon": "mdi-bat",
+	"description": "Creates a new bat entity.",
+	"category": "fileType.entity",
+	"targetVersion": [">=", "1.8.0"],
+	"fields": [
+		["Identifier", "IDENTIFIER"],
+		["Display Name", "IDENTIFIER_NAME"]
+	],
+
+	"createFiles": [
+		[
+			"entity.json",
+			"BP/entities/{{IDENTIFIER}}.json",
+			{ "inject": ["IDENTIFIER", "PROJECT_PREFIX"] }
+		],
+		[
+			"animation.json",
+			"RP/animations/{{IDENTIFIER}}.json",
+			{ "inject": ["IDENTIFIER"] }
+		]
+	],
+	"expandFiles": [
+		[
+			"en_US.lang",
+			"RP/texts/en_US.lang",
+			{ "inject": ["IDENTIFIER", "IDENTIFIER_NAME", "PROJECT_PREFIX"] }
+		]
+}
+```
+
+### Variable Example
+
+```javascript
+{
+  "texture_data": {
+    "{{IDENTIFIER}}": {
+      "textures": [ "textures/items/{{IDENTIFIER}}" ]
+    }
+  }
+}
+```
+
+### Field Exaples
+
+```javascript
+["Text", "TEXT"]
+```
+
+```javascript
+[
+	"File", 
+	"FILE",
+	{
+		"type": "fileInput",
+		"accept": "image/png"
+	}
+]
+```
+
+```javascript
+[
+	"Selection box",
+	"SELECT",
+	{
+		"type": "selectInput",
+		"default": "select1",
+		"options": [
+			{"text": "select1", "value": "select1"},
+			{"text": "select2", "value": "select2"},
+			{"text": "select3", "value": "select3"}
+		]
+	}
+]
+```
+
+```javascript
+[
+	"Number Slider",
+	"NUM_SLIDE",
+	{
+		"type": "numberInput",
+		"min": 0,
+		"max": 10,
+		"step": 1
+	}
+]
+```
+
+```javascript
+[
+	"Switch me!",
+	"BOOLEAN",
+	{
+		"type": "switch",
+	}
+]
+```
+
+### More Examples:
+
+
+-	[`Bridge. Vannila Presets`](https://github.com/bridge-core/editor/tree/main/data/preset)
+-	[`More Vannila Entity Presets by Joelant05`](https://github.com/bridge-core/plugins/tree/master/plugins/MoreVanillaEntityPresets)
+
+
+
+Page written by Paty
```

Comment:
The author tag on our website is working automatically because GitHub knows that you have written the file. You can delete this line :)